### PR TITLE
Improved exception handling for received CIM-XML in tuple parser

### PR DESCRIPTION
**Pull request by andy-maier**
See commit message.

I don't think this PR needs to be rolled back into pywbem 1.1.

### CODE REVIEW

File: tests/unittest/pywbem/test_indicationlistener.py [hunk 1]
```diff
@@ -988,7 +988,7 @@ def test_WBEMListener_incorrect_headers(desc, headers, exp_status, exp_headers):
         </CIM>""",
         400,
         {
-            'CIMErrorDetails': r'Element .CIM. missing required '
+            'CIMErrorDetails': r'Element .CIM. does not have required '
```
Comment (KSchopmeyer) [inline tests/unittest/pywbem/test_indicationlistener.py:991]: Actually I would prefer the original text with the word missing to "does not have". There are several cases of this

File: pywbem/_tupleparse.py [hunk 2]
```diff
@@ -2015,7 +2167,7 @@ def parse_methodcall(self, tup_tree):
                                      ('LOCALCLASSPATH', 'LOCALINSTANCEPATH'))
         if not path:
             raise CIMXMLParseError(
-                _format("Element {0!A} missing a required child element "
+                _format("Element {0!A} does not have required child elements "
```
Comment (KSchopmeyer) [inline pywbem/_tupleparse.py:2170]: Actually I prefer"missing" to "does not have"

File: pywbem/_tupleparse.py [hunk 3]
```diff
@@ -1987,7 +2139,7 @@ def parse_imethodcall(self, tup_tree):
 
         if not k:
             raise CIMXMLParseError(
-                _format("Element {0!A} missing child elements "
+                _format("Element {0!A} does not have any child elements "
```
Comment (KSchopmeyer) [inline pywbem/_tupleparse.py:2142]: See comment below about replaciing word "missing"

File: pywbem/_tupleparse.py [hunk 4]
```diff
@@ -150,13 +150,14 @@ def pcdata(self, tup_tree):
         """
         try:
             data = u''.join(tup_tree[2])
-
         except TypeError:
-            raise CIMXMLParseError(
+            new_exc = CIMXMLParseError(
```
Comment (KSchopmeyer) [inline pywbem/_tupleparse.py:154]: This one confuses me in that I thought that __cause__ was only involves when the
raise was of the form

raise new_exc from original_exc

File: tests/unittest/pywbem/test_indicationlistener.py (hunk 1 above)
Comment (andy-maier) [reply tests/unittest/pywbem/test_indicationlistener.py:991]: Missing is an emotion. Does the element really miss its kids?

File: pywbem/_tupleparse.py (hunk 4 above)
Comment (andy-maier) [reply pywbem/_tupleparse.py:154]: It is the portable and more recent alternative to that.

File: tests/unittest/pywbem/test_indicationlistener.py (hunk 1 above)
Comment (andy-maier) [reply tests/unittest/pywbem/test_indicationlistener.py:991]: We agreed to say "is missing"

Comment (andy-maier) [reply tests/unittest/pywbem/test_indicationlistener.py:991]: DONE

File: pywbem/_tupleparse.py (hunk 2 above)
Comment (andy-maier) [reply pywbem/_tupleparse.py:2170]: DONE - changed to "is missing"

File: pywbem/_tupleparse.py (hunk 3 above)
Comment (andy-maier) [reply pywbem/_tupleparse.py:2142]: DONE - changed to "is missing required"

**Comment by andy-maier** [comment]
Ready for a final review.

**Review by KSchopmeyer — APPROVED** [review]

**Pull request merged**